### DISCUSSION TITLE
ROI trigger for looming experiment

### DISCUSSION THREAD

**Discussion by kmelani** [bonsai-rx/bonsai#1441] [Q&A]
Hi! i'm working on coding an ROI trigger to be used for my sweeping/looming experiment. i've been looking at the ROI example offered in the gallery, and I'm a little confused on how I need to edit it for what I need it to do. essentially, i need to change it so that my sweep/loom stimulus is triggered when the mouse enters the middle of my arena (tracked by the camera)
any direction would be very appreciated. thank you in advance!

**Comment by bruno-f-cruz** [comment]
Can you share your current workflow? Thanks

**Reply by kmelani** [reply]
[LOOMING STIMULUS.zip](https://github.com/bonsai-rx/bonsai/files/11909827/LOOMING.STIMULUS.zip)
[sweep stimulus.zip](https://github.com/bonsai-rx/bonsai/files/11909828/sweep.stimulus.zip)
hi! these are my visual stimulus codes. i would need to do an roi trigger for both. i haven't been able to find any sources to start, other than the roi trigger example in the bonsai gallery for an roi trigger for a sound stimulus.

**Reply by kmelani** [reply]
[sweep with roi in works.zip](https://github.com/bonsai-rx/bonsai/files/11910240/sweep.with.roi.in.works.zip)
here we are trying to get the roi working to this video just so we can get it working. whenever i connect the pathway to the subscribewhen, it still does not time the trigger right

**Reply by bruno-f-cruz** [reply]
You need to connect a trigger to the `SubscribeWhen`. Maybe a `PusblishSubject` from the output of your condition?

**Reply by kmelani** [reply]
i added a publishsubject to the end of the code, connecting it to conditions. when i connect the publishsubject to the subscribewhen, it still doesnt trigger the sweep when the mouse enters the roi.

**Reply by bruno-f-cruz** [reply]
Can you share the workflow?

**Reply by kmelani** [reply]
[sweep with roi in works update 1.zip](https://github.com/bonsai-rx/bonsai/files/11916537/sweep.with.roi.in.works.update.1.zip)

**Reply by kmelani** [reply]
hi! i played around with what i had more and it seems like i got the stimulus to be triggered by something in the roi region of interest, yet i think its being triggered by the wire rather than the mouse (since that ends up entering the region). is there any way i can get it to ignore the wire? the idea is that the sweeping stimulus shouldn't be triggered until the mouse enters the region of interest only.
[sweep with roi in works update 2.zip](https://github.com/bonsai-rx/bonsai/files/11918155/sweep.with.roi.in.works.update.2.zip)

**Reply by chiyu1203** [reply]
when you said wire, did you mean a wire attached to the animal? In this case you probably need to set a threshold for the area of detection so that wire is too small to trigger the condition. I am not entire sure how you detect animal entering the ROI but here is an example to set a threshold for such a purpose. Hope this can solve your problem. Otherwise, you might want to go for bonsai + real time pose estimation
[entering_roi_example.zip](https://github.com/bonsai-rx/bonsai/files/11940672/entering_roi_example.zip)

**Reply by kmelani** [reply]
hi! thank you, ill definitely try that.
to try and explain the roi a little more: we will be performing our looming experiment on mice. the mice will have a mini m-scope on their skulls to track cortical outputs of calcium. this camera for imaging is going to be attached above the mouse arena and their head. the issue is, we'd like to make an roi trigger to trigger our visual stimulus (the sweeping and then looming stimuli) when the mouse enters the roi threshold. sometimes, the wire seems to drift into the roi trigger without the mouse, since it's attached to an anchor above the arena. so the issue has been trying to draw an roi that only detects the mouse to trigger the stimulus.
thanks for your input and roi example!
